cache-config/testing/ort-tests/t3c-lockfile_test.go

```diff
@@ -0,0 +1,113 @@
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import (
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/apache/trafficcontrol/cache-config/testing/ort-tests/tcdata"
+)
+
+func TestLockfile(t *testing.T) {
+	testName := "TestLockfile"
+	t.Logf("------------- Starting " + testName + " ---------------")
+	tcd.WithObjs(t, []tcdata.TCObj{
+		tcdata.CDNs, tcdata.Types, tcdata.Tenants, tcdata.Parameters,
+		tcdata.Profiles, tcdata.ProfileParameters, tcdata.Statuses,
+		tcdata.Divisions, tcdata.Regions, tcdata.PhysLocations,
+		tcdata.CacheGroups, tcdata.Servers, tcdata.Topologies,
+		tcdata.DeliveryServices}, func() {
+
+		hostName := "atlanta-edge-03"
+		const fileName = `records.config`
+
+		firstOut := ""
+		firstCode := 0
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			t.Logf("TestLockFile first t3c starting %v", time.Now())
+			firstOut, firstCode = t3cUpdateReload(hostName, "badass")
+			t.Logf("TestLockFile first t3c finished %v", time.Now())
+		}()
+
+		time.Sleep(time.Millisecond * 100) // sleep long enough to ensure the concurrent t3c starts
+		t.Logf("TestLockFile second t3c starting %v", time.Now())
+		out, code := t3cUpdateReload(hostName, "badass")
+		t.Logf("TestLockFile second t3c finished %v", time.Now())
+		if code != 0 {
+			t.Fatalf("second t3c apply badass failed: output '''%v''' code %v\n", out, code)
+		}
+
+		wg.Wait()
+		if firstCode != 0 {
+			t.Fatalf("first t3c apply badass failed: output '''%v''' code %v\n", firstOut, firstCode)
+		}
+
+		outStr := string(out)
+		outLines := strings.Split(outStr, "\n")
+		acquireStartLine := ""
+		acquireEndLine := ""
+		for _, line := range outLines {
+			if strings.Contains(line, "Trying to acquire app lock") {
+				acquireStartLine = line
+			} else if strings.Contains(line, "Acquired app lock") {
+				acquireEndLine = line
+			}
+			if acquireStartLine != "" && acquireEndLine != "" {
+				break
+			}
+		}
+
+		if acquireStartLine == "" || acquireEndLine == "" {
+			t.Fatalf("t3c apply output expected to contain 'Trying to acquire app lock' and 'Acquired app lock', actual: '''%v'''", out)
+		}
+
+		acquireStart := parseLogLineTime(acquireStartLine)
+		if acquireStart == nil {
+			t.Fatalf("t3c apply acquire line failed to parse time, line '" + acquireStartLine + "'")
+		}
+		acquireEnd := parseLogLineTime(acquireEndLine)
+		if acquireEnd == nil {
+			t.Fatalf("t3c apply acquire line failed to parse time, line '" + acquireEndLine + "'")
+		}
+
+		minDiff := time.Second * 1 // checking the file lock should never take 1s, so that's enough to verify it was hit
+		if diff := acquireEnd.Sub(*acquireStart); diff < minDiff {
+			t.Fatalf("t3c apply expected time to acquire while another t3c is running to be at least %v, actual %v start line '%v' end line '%v'", minDiff, diff, acquireStartLine, acquireEndLine)
+		}
+
+		t.Logf(testName + " succeeded")
```

Comment:
Is this necessary? The output of `go test` informs you whether the test succeeded or failed

Reply:
I added that line to distinguish the `tcd.WithObjs(t, []tcdata.TCObj{` block getting to the end successfully, versus the outer test itself. I found it very useful when working on the test. I can remove it; but it exists because it was very useful in-practice writing, fixing, and figuring out why and where the test was failing.

Reply:
I'm not sure I understand the use-case, but it's not a big deal - and any rate there are already some extraneous log lines, so it may just be best addressed all at once.